### PR TITLE
Improve engine log error

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/26013|

# Objective

This PR aims to change the `std::cerr` current output, for the login approach which is splog.

We apply only to source code inside of the `engine` folder, excluding:

- Dependencies
- UTs
- TesTools

### CODE REVIEW

**Review by sebasfalcone — CHANGES_REQUESTED** [review]
Missing files:
- src/shared_modules/utils/epollWrapper.hpp
- src/shared_modules/utils/socketClient.hpp
- src/shared_modules/utils/socketServer.hpp
- src/shared_modules/utils/socketWrapper.hpp
- src/shared_modules/utils/threadDispatcher.h

I've excluded testools and tests

File: src/engine/source/base/include/base/utils/threadEventDispatcher.hpp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 #ifndef _THREAD_EVENT_DISPATCHER_HPP
 #define _THREAD_EVENT_DISPATCHER_HPP
 
+#include "base/logging.hpp"
```
Comment (juliancnn) [inline src/engine/source/base/include/base/utils/threadEventDispatcher.hpp:15]: Should be:
```cpp
#include <atomic>
#include <iostream>
#include <thread>

#include <base/logging.hpp>

#include "rocksDBQueue.hpp"
#include "threadSafeQueue.hpp"
```

The style is:
```
#include <STD>
#include <STD>
#include <STD>

#include <System>
#include <System>

#include <External-lib>
#include <External-lib>

#include <internal-module>
#include <internal-module>

#include "<local/module>"

#include "same-folder"
```

**Review by juliancnn — CHANGES_REQUESTED** [review]
Discard changes on `cmds/src/*` and `main.cpp`, the error should be printed in the stderr because it is not possible to know if the logger is initialized and even if it is necessary, since for the clients the logger should not be used, only for the cmd `start`.

File: src/engine/source/base/include/base/utils/threadEventDispatcher.hpp (hunk 1 above)
Comment (GabrielEValenzuela) [reply src/engine/source/base/include/base/utils/threadEventDispatcher.hpp:15]: Done!

**Review by sebasfalcone — DISMISSED** [review]

**Review by juliancnn — CHANGES_REQUESTED** [review]
UT should be fixed

**Comment by GabrielEValenzuela** [comment]
I made a rebase and run tests locally, they are ok

```console
[----------] 6 tests from ThreadEventDispatcherTest
[ RUN      ] ThreadEventDispatcherTest.ConstructorTestSingleThread
[       OK ] ThreadEventDispatcherTest.ConstructorTestSingleThread (5079 ms)
[ RUN      ] ThreadEventDispatcherTest.ConstructorTestMultiThread
[       OK ] ThreadEventDispatcherTest.ConstructorTestMultiThread (5062 ms)
[ RUN      ] ThreadEventDispatcherTest.ConstructorTestMultiThreadDifferentTypeExceptions
[       OK ] ThreadEventDispatcherTest.ConstructorTestMultiThreadDifferentTypeExceptions (28 ms)
[ RUN      ] ThreadEventDispatcherTest.ConstructorTestMultiThreadDifferentTypeExceptionsDropElement
operator() ERROR: event discarded: 0. Unexpected error during event processing.
operator() ERROR: event discarded: 20. Unexpected error during event processing.
operator() ERROR: event discarded: 40. Unexpected error during event processing.
operator() ERROR: event discarded: 60. Unexpected error during event processing.
operator() ERROR: event discarded: 80. Unexpected error during event processing.
[       OK ] ThreadEventDispatcherTest.ConstructorTestMultiThreadDifferentTypeExceptionsDropElement (5025 ms)
[ RUN      ] ThreadEventDispatcherTest.CtorNoWorkerSingleThread
[       OK ] ThreadEventDispatcherTest.CtorNoWorkerSingleThread (5062 ms)
[ RUN      ] ThreadEventDispatcherTest.CtorPopFeatureSingleThread
2024-10-03 15:57:05.896 584767:585117 error: Dispatch handler error: Test exception
[       OK ] ThreadEventDispatcherTest.CtorPopFeatureSingleThread (32 ms)
[----------] 6 tests from ThreadEventDispatcherTest (20291 ms total)
```

**Comment by juliancnn** [comment]
> I made a rebase and run tests locally, they are ok
> 
> ```
> [----------] 6 tests from ThreadEventDispatcherTest
> [ RUN      ] ThreadEventDispatcherTest.ConstructorTestSingleThread
> [       OK ] ThreadEventDispatcherTest.ConstructorTestSingleThread (5079 ms)
> [ RUN      ] ThreadEventDispatcherTest.ConstructorTestMultiThread
> [       OK ] ThreadEventDispatcherTest.ConstructorTestMultiThread (5062 ms)
> [ RUN      ] ThreadEventDispatcherTest.ConstructorTestMultiThreadDifferentTypeExceptions
> [       OK ] ThreadEventDispatcherTest.ConstructorTestMultiThreadDifferentTypeExceptions (28 ms)
> [ RUN      ] ThreadEventDispatcherTest.ConstructorTestMultiThreadDifferentTypeExceptionsDropElement
> operator() ERROR: event discarded: 0. Unexpected error during event processing.
> operator() ERROR: event discarded: 20. Unexpected error during event processing.
> operator() ERROR: event discarded: 40. Unexpected error during event processing.
> operator() ERROR: event discarded: 60. Unexpected error during event processing.
> operator() ERROR: event discarded: 80. Unexpected error during event processing.
> [       OK ] ThreadEventDispatcherTest.ConstructorTestMultiThreadDifferentTypeExceptionsDropElement (5025 ms)
> [ RUN      ] ThreadEventDispatcherTest.CtorNoWorkerSingleThread
> [       OK ] ThreadEventDispatcherTest.CtorNoWorkerSingleThread (5062 ms)
> [ RUN      ] ThreadEventDispatcherTest.CtorPopFeatureSingleThread
> 2024-10-03 15:57:05.896 584767:585117 error: Dispatch handler error: Test exception
> [       OK ] ThreadEventDispatcherTest.CtorPopFeatureSingleThread (32 ms)
> [----------] 6 tests from ThreadEventDispatcherTest (20291 ms total)
> ```

The test fails so that the logger is not initialized for the module. Remember that ctest has 2 binaries for the tests. You can run them with the test main or just the test binary. If you run the test binary you are not initializing the module.

File: src/engine/source/base/test/src/unit/utils/threadEventDispatcher_test.cpp [hunk 2]
```diff
@@ -9,6 +9,7 @@
  * Foundation.
  */
 
+#include "base/logging.hpp"
```
Comment (sebasfalcone) [inline src/engine/source/base/test/src/unit/utils/threadEventDispatcher_test.cpp:12]: Given that we are modifying this file, let's re-order the headers based on the codding style proposed

**Review by sebasfalcone — CHANGES_REQUESTED** [review]

File: src/engine/source/base/test/src/unit/utils/threadEventDispatcher_test.cpp (hunk 2 above)
Comment (GabrielEValenzuela) [reply src/engine/source/base/test/src/unit/utils/threadEventDispatcher_test.cpp]: Done!

File: src/engine/source/base/test/src/unit/utils/threadEventDispatcher_test.cpp [hunk 3]
```diff
@@ -9,14 +9,16 @@
  * Foundation.
  */
 
-#include "base/utils/rocksDBQueue.hpp"
-#include "base/utils/threadEventDispatcher.hpp"
-#include "base/utils/threadSafeMultiQueue.hpp"
 #include <future>
 #include <gtest/gtest.h>
 #include <memory>
 #include <thread>
```
Comment (juliancnn) [inline src/engine/source/base/test/src/unit/utils/threadEventDispatcher_test.cpp:16]:
```suggestion
#include <future>
#include <memory>
#include <thread>

#include <gtest/gtest.h>
```

Comment (juliancnn) [reply src/engine/source/base/test/src/unit/utils/threadEventDispatcher_test.cpp:16]: Minor change

**Review by juliancnn — CHANGES_REQUESTED** [review]

File: src/engine/source/base/test/src/unit/utils/threadEventDispatcher_test.cpp (hunk 3 above)
Comment (GabrielEValenzuela) [reply src/engine/source/base/test/src/unit/utils/threadEventDispatcher_test.cpp]: Done!

**Review by juliancnn — CHANGES_REQUESTED** [review]
LGTM!

**Review by juliancnn — APPROVED** [review]
LGTM!

**Review by sebasfalcone — APPROVED** [review]

**Pull request merged**